js/interface.js

```diff
@@ -307,32 +311,115 @@
     return Fliplet.API.request({
       url: [
         'v1/widgets?include_instances=true&tags=type:menu',
+        '&include_all_versions=true',
         '&appId=' + Fliplet.Env.get('appId'),
-        '&organizationId=' + (Fliplet.Env.get('organizationId') || '')
+        '&organizationId=' + (organizationId || '')
       ].join('')
     }).then(function(response) {
       return Promise.resolve(response.widgets);
     });
   }
 
+  /**
+   * Function to filter organizationMenus by version.
+   * @param {Array} data - array of organizationMenus
+   * @param {Object} latestVersionMenu - current active menu
+   * @returns {Object} latestVersionMenu - object which contains the latest version of current menu package type
+   */
+
+  function sortByVersion(data, latestVersionMenu) {
+    var previousVersion;
+
+    data.forEach(function(menu) {
+      // Compare each record with the previous one by version to find the newest version.
+      if (previousVersion && previousVersion.version !== menu.version) {
+        latestVersionMenu = (previousVersion.version > menu.version) ? previousVersion : menu;
+
+        return;
+      }
+
+      previousVersion = menu;
+
+      // Always write the system menu first, then if iteration finds the newest version, latestVersionMenu will be overwritten.
+      if (!latestVersionMenu) {
+        latestVersionMenu = menu;
+      }
+    });
+
+    return latestVersionMenu;
+  }
+
+  /**
+   * Creates a list of menu widgets to be displayed based on versions and current usage
+   * @param {Array} menus - List of menu widgets
+   * @returns {Array} List of menu widgets to be displayed based on versions and current usage
+   */
+
+  function generateMenuList(menus) {
+    var result = [];
+    var hasActiveMenu = false;
+
+    // System menus don't have organizationId
+    var systemMenus = menus.filter(function(item) {
```

Comment:
I see you've generated `systemMenus`, then you looped through these to generate the correct menu version based on usage and version.

How about the organisation menus, i.e. the menus with `organizationId` assigned? They need to be displayed too and displayed based on the same logic. Currently it looks like they are being ignored entirely.

Reply:
@tonytlwu, fixed